### ISSUE TITLE
mdmctl mdmcert.download http request failure

### ISSUE THREAD

**Issue by n8felton** [micromdm/micromdm#463]
### What version of `mdmctl` are you using?
```
$ mdmctl version
mdmctl version v1.3.1
```

### What `mdmctl` command did you run?
```
$ mdmctl mdmcert.download -new -email=<email registered and verified>
```

### What did you expect to see?
```
Request successfully sent to mdmcert.download. Your CSR should now
be signed. Check your email for next steps. Then use the -decrypt option
to extract the CSR request which will then be uploaded to Apple.
```
(Following directions at https://github.com/micromdm/micromdm/wiki/mdmcert.download)

### What did you see instead?
```
error sending http request: got unexpected result body: "failure"
```

Despite the error message, these files were still generated:
```
mdmcert.download.pki.crt
mdmcert.download.pki.key
mdmcert.download.push.csr
mdmcert.download.push.key
```

**Comment by n8felton** [comment]
Ok, I think I found the cause...

When I registered my email address, I used `Nathan.Felton`, but when I ran the command, I used `nathan.felton`. I ran the command again using `Nathan.Felton`, and it appears to be working.

TL;DR - Email address appears to be case sensitive.

**Comment by jessepeterson** [comment]
Ah, interesting. Yes looking at the mdmcert.download code that makes sense. We don't store any information on disk unencrypted/unhashed. And we're only hashing the raw input email address and validating against that. That was a hasty decision!

At the very least we should improve the error handling to give a better response than just "failure." Glad it otherwise worked, though!
